packages/wrangler/src/pages/deployment-tails.tsx

```diff
@@ -137,34 +138,54 @@ export async function Handler({
 		PAGES_CONFIG_CACHE_FILENAME
 	);
 	const accountId = await requireAuth(pagesConfig);
+	let deploymentId = deployment;
 
 	projectName = projectName || pagesConfig.project_name;
 
-	if (!deploymentId && !projectName && isInteractive) {
+	// When not specifying a deployment, we need a project name to use the latest
+	// Or when specifying by deployment URL
+	if ((!deployment || isUrl(deployment)) && !projectName && isInteractive) {
 		projectName = await promptSelectProject({ accountId });
 	}
 
-	if (!deploymentId && !projectName) {
-		throw new FatalError("Must specify a project name or deployment ID.", 1);
+	if (!deployment && !projectName) {
+		throw new FatalError("Must specify a project name or deployment.", 1);
 	}
 
-	if (!deploymentId) {
-		logger.log(
-			"No deploymentId specified. Using latest deployment for ",
-			environment,
-			"environment."
-		);
+	const deployments: Array<Deployment> = await fetchResult(
+		`/accounts/${accountId}/pages/projects/${projectName}/deployments`
+	);
 
-		const deployments: Array<Deployment> = await fetchResult(
-			`/accounts/${accountId}/pages/projects/${projectName}/deployments`
-		);
+	const envDeployments = deployments.filter(
+		(d) => d.environment === environment
+	);
 
-		const envDeployments = deployments.filter(
-			(d) => d.environment === environment
+	if (envDeployments.length === 0) {
+		throw new FatalError("No deployments for environment: " + environment, 1);
+	}
+
+	// Deployment is URL
+	if (deployment?.startsWith("https://")) {
+		const { hostname: deploymentHostname } = new URL(deployment);
+		const targetDeployment = envDeployments.find(
+			(d) => new URL(d.url).hostname === deploymentHostname
 		);
 
-		if (envDeployments.length === 0) {
-			throw new FatalError("No deployments for environment: " + environment, 1);
+		if (!targetDeployment) {
+			throw new FatalError(
+				"Could not find deployment match url: " + deployment,
+				1
+			);
+		}
+
+		deploymentId = targetDeployment.id;
+	} else {
+		if (format === "pretty") {
+			logger.log(
+				"No deployment specified. Using latest deployment for ",
```

Comment:
remove white space after `for`. It's not necessary 👇 . Or perhaps use `No deployment specified. Using latest deployment for ${environment} environment.`

<img width="635" alt="Screenshot 2022-11-11 at 11 52 54" src="https://user-images.githubusercontent.com/4638332/201325776-998dda00-f000-4006-b974-3f16f47e04a0.png">